search.go

```diff
@@ -105,26 +111,49 @@ type FacetRequest struct {
 }
 
 func (fr *FacetRequest) Validate() error {
-	if len(fr.NumericRanges) > 0 && len(fr.DateTimeRanges) > 0 {
+	nrCount := len(fr.NumericRanges)
+	drCount := len(fr.DateTimeRanges)
+	if nrCount > 0 && drCount > 0 {
 		return fmt.Errorf("facet can only conain numeric ranges or date ranges, not both")
 	}
-
-	nrNames := map[string]interface{}{}
-	for _, nr := range fr.NumericRanges {
-		if _, ok := nrNames[nr.Name]; ok {
-			return fmt.Errorf("numeric ranges contains duplicate name '%s'", nr.Name)
+	rangeSet := false
+	if nrCount > 0 {
+		nrNames := map[string]interface{}{}
+		for _, nr := range fr.NumericRanges {
+			if _, ok := nrNames[nr.Name]; ok {
+				return fmt.Errorf("numeric ranges contains duplicate name '%s'", nr.Name)
+			}
+			nrNames[nr.Name] = struct{}{}
+			if !rangeSet && (nr.Min != nil || nr.Max != nil) {
+				rangeSet = true
+			}
+		}
+		if !rangeSet {
+			return fmt.Errorf("numeric range query must specify either min, max or both")
```

Comment:
Yes Marty, I wasn't sure of all the valid facet nr/dr formats. So does it mean the above mentioned query would like below Or I unable to envision how  multiple numeric ranges works against a single field.
```
curl -XPOST -H "Content-Type: application/json" \
 http://127.0.0.1:9201/api/index/idx/query \
 -d '{
  "size": 10,
  "query": {
    "boost": 1,
    "query": "water"
  },
  "facets": {
    "abv": {
      "size": 5,
      "field": "abv",
      "numeric_ranges": [
        {
          "name": "high",
          "max1": 7
        },
        {
          "name": "low",
          "min1": 7
        }
      ]
    },
    "abv_second": {
      "size": 15,
      "field": "abv",
      "numeric_ranges": [
        {
          "name": "high",
          "max1": 7
        },
        {
          "name": "low",
          "min1": 7
        }
      ]
    }
  }
}'
```